test/api/fixmarkup.js

```diff
@@ -5,11 +5,21 @@ const hljs   = require('../../build');
 
 describe('.fixmarkup()', () => {
   after(() => {
-    hljs.configure({ useBR: false })
-  })
+    hljs.configure({ useBR: false });
+  });
+
+  it('should not strip HTML from beginning of strings', () => {
+    hljs.configure({ useBR: true });
+    const value = '<span class="hljs-attr">"some"</span>: \n <span class="hljs-string">"json"</span>';
```

Comment:
the issue occurs when there is no any `\n` in the original string.
So test config looks like this: 
```
 hljs.configure({ useBR: true });
    const value = '<span class="hljs-attr">"some"</span>';
```

Reply:
Yes, because the regex matches against ^... so any content (just not a newline) would prevent you from seeing this bug.  What I need to know is WHY we have this behavior in the first place - because it looks very intentional.